### PR TITLE
Update InferCNV: control number of threads, downgrade profile and sparse matrix support

### PR DESCRIPTION
**Pull request by pcm32**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [x] This PR does something else (explain below)

Minor change in the wrapper to control number of threads. It also downgrades the profile for wider compatibility.

### CODE REVIEW

**Comment by pcm32** [comment]
I will test installation as is first to see if the API based installation with higher profile than the instance works.

File: tools/infercnv/infercnv.xml [hunk 1]
```diff
@@ -53,6 +55,9 @@
                                 num_ref_groups = NULL, # keep this as default
                                 ref_subtract_use_mean_bounds = TRUE, # keep this as default
 
+                                # Control number of threads for multiprocessesor use
+                                #num_threads = num_threads_galaxy_slots,
```
Comment (bgruening) [inline tools/infercnv/infercnv.xml:59]:
```suggestion
                                #num_threads = num_threads_galaxy_slots,
```

You need to put the `os.environ.get('GALAXY_SLOTS', 1)` into this script.

Comment (pcm32) [reply tools/infercnv/infercnv.xml:59]: You mean I should grab the GALAXY_SLOT from R at level with `Sys.getenv('GALAXY_SLOT')` or use the cheetah construct `$os.environ.get('GALAXY_SLOTS', 1)` directly here? Won't I still need the cheetah `import os` which might be breaking things? Thanks.

Comment (bgruening) [reply tools/infercnv/infercnv.xml:59]: You can not use cheetah, you need to get the envs from within the script. In that case your R script.

Comment (pcm32) [reply tools/infercnv/infercnv.xml]: I see... I found a potential way in another tool, tried before I saw your response, if it doesn't work will switch to the R way.

**Comment by pcm32** [comment]
...finally....

File: tools/infercnv/infercnv.xml (hunk 1 above)
Comment (pcm32) [reply tools/infercnv/infercnv.xml]: Moved now to full R logic for grabbing GALAXY_SLOTS. Thanks!

**Comment by pcm32** [comment]
Looking in more detail, this tool is receiving a non-sparse matrix in tsv format, that is prohibitive for large datasets. The tool should instead receive a sparse matrix format, ideally matrix market or a seurat object. Otherwise this means exploding very large matrices.

**Comment by bgruening** [comment]
LGTM. Do you want to add the matrix-market ...?

**Comment by pcm32** [comment]
Yes, almost there with the sparse matrix support.

**Comment by pcm32** [comment]
I'm once again somehow offending the Cheetah gods... 

```
Job in error state.. tool_id: infercnv, exit_code: None, stderr: .
Job in error state.. tool_id: infercnv, exit_code: None, stderr: .
Output collection 'infercnv_results': failed to find identifier 'infercnv.heatmap_thresholds' in the tool generated elements []
```

new test passes, but previous ones don't - could you please advice @bgruening ? Thanks.

File: tools/infercnv/infercnv.xml [hunk 2]
```diff
@@ -9,6 +9,12 @@
             ## extract gene order from gtf file
             python '$__tool_directory__/gtf_to_position_file.py' --attribute_name $attribute '$gtf_file' 'gene_order.tabular' &&
         #end if
+        #if $input_mat.mtx and $input_mat.mtx_genes and $input_mat.mtx_barcodes:
```
Comment (nilchia) [inline tools/infercnv/infercnv.xml:12]:
```suggestion
        #if str($input_mat.format) == "mtx":
```

File: tools/infercnv/infercnv.xml [hunk 3]
```diff
@@ -19,9 +25,16 @@
             # load library
             library('infercnv')
 
+            #if $input_mat.mtx and $input_mat.mtx_genes and $input_mat.mtx_barcodes:
```
Comment (nilchia) [inline tools/infercnv/infercnv.xml:28]:
```suggestion
            #if str($input_mat.format) == "mtx":
```

**Review by nilchia — CHANGES_REQUESTED** [review]

**Comment by pcm32** [comment]
Thanks @nilchia for the suggestions!

**Comment by nilchia** [comment]
You're welcome @pcm32 !
Thanks for the update.

**Review by nilchia — APPROVED** [review]

File: tools/infercnv/infercnv.xml [hunk 4]
```diff
@@ -169,7 +190,20 @@
...
+            <param name="format" type="select" label="Matrix format">
+                <option value="tsv">Tabular (non-sparse matrix)</option>
+                <option value="mtx" selected="true">10x sparse matrix</option>
+            </param>
+            <when value="tsv">
+                <param name="raw_mtx" type="data" format="tabular" label="Raw counts matrix" help="The matrix of genes (rows) vs. cells (columns) containing the raw counts" />
+            </when>
+            <when value="mtx">
+                <param name="mtx" type="data" format="mtx,tabular" label="Raw counts matrix (matrix market)" help="The matrix of genes (rows) vs. cells (columns) containing the raw counts" />
+                <param name="mtx_genes" type="data" format="tsv,tabular" label="Genes file" help="The genes file for matrix market matrix, a two column symbol - symbol or gene_id - symbol tab separated file" />
```
Comment (bgruening) [inline tools/infercnv/infercnv.xml:203]: tabular and tsv? I guess tabular is enough: https://github.com/galaxyproject/galaxy/blob/e9c6a071c73ec764ddb0963c87b0e4e829138872/lib/galaxy/datatypes/tabular.py#L1512

Comment (pcm32) [reply tools/infercnv/infercnv.xml]: Agreed. Added txt in exchange, sometimes on upload some of those files end up as text and then is less hassle to use the tool/pipeline.

**Review by bgruening — APPROVED** [review]

**Pull request merged**